core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -1111,6 +1111,10 @@ class ZKCheckedEphemeral(path: String,
                             case Code.INVALIDACL =>
                               error("Invalid ACL")
                               setResult(Code.INVALIDACL)
+                            case Code.NOAUTH =>
+                              // Allows old consumer to register with secure ZK
+                              error("NOAUTH raised on path %s (AUTH may not be needed to create children)".format(path))
+                              createRecursive(path, suffix)
```

Comment:
> Items 1 and 2 I believe will be caught by the CreateCallback and/or GetDataCallback. I think logging the error handles troubleshooting issues.

Hmm... Let me see if I understand your suggestion. If the parent node creation fails with NOAUTH and the node doesn't exist, then we will proceed anyway to attempt to create the child node and fail with some different error. Then you need to use the logs to figure out what has actually happened. Is that correct?

It's not the crispest function contract, is it? In truth, I'm a little surprised to see you suggest an approach which makes troubleshooting more difficult. The only thing we get out of it is the skipped existence check? Does not seem worth it to me.